Emby.Server.Implementations/SystemManager.cs

```diff
@@ -72,16 +76,32 @@ public SystemInfo GetSystemInfo(HttpRequest request)
         };
     }
 
-    /// <inheritdoc />
+    /// <summary>
+    /// Gets public system info of the server.
+    /// LocalAddress is the server URL that can also be used by applications not on the server, for instance
+    /// a cast receiver device. In case the request has a localhost url, LocalAddress will be set to a
+    /// server Private network address (IPv4) or Unique Local Address (ULA, IPv6) which is valid for all
+    /// devices on the LAN.
+    /// </summary>
+    /// <param name="request">Where this request originated.</param>
+    /// <returns>PublicSystemInfo.</returns>
     public PublicSystemInfo GetPublicSystemInfo(HttpRequest request)
     {
+        string host = request.Host.ToString();
+        string localAddress = _applicationHost.GetSmartApiUrl(request);
+
+        if (host.StartsWith("localhost:", StringComparison.OrdinalIgnoreCase) || host.StartsWith("[::1]", StringComparison.OrdinalIgnoreCase) || host.StartsWith("127.0", StringComparison.OrdinalIgnoreCase))
```

Comment:
We should have Subnet specifications for those networks inside the network extensions which you can check against, this should be more reliable.

Reply:
Does this look better?